command/agent/csi_endpoint.go

```diff
@@ -57,11 +62,11 @@ func (s *HTTPServer) CSIVolumeSpecificRequest(resp http.ResponseWriter, req *htt
 	id := tokens[0]
 
 	switch req.Method {
-	case "GET":
+	case http.MethodGet:
 		return s.csiVolumeGet(id, resp, req)
-	case "PUT":
-		return s.csiVolumePut(id, resp, req)
-	case "DELETE":
+	case http.MethodPut:
```

Comment:
If we're allowing `POST` in `csiVolumePut` should we allow it here as well?

Reply:
Well, we don't have to allow `POST` in the other, our rest conventions are a little weird around here. The method is missing `id` because our raft messages can send multiple volumes in the body, and passing the id is redundant. It also allows just using the same method for `POST`. If we're sticking close to typical rest conventions, `PUT` is for changing an object you've specified in the url and `POST` is directed at the parent and produces children as a side-effect. `csiVolumePut` allows `POST` so that `POST` can be supported up in `CSIVolumesRequest` where it could be used with multiple volumes.

Reply:
Sounds good.